src/Datastore/DatastoreClient.php

```diff
@@ -0,0 +1,907 @@
+    {
+        $options = $options + [
+            'namespaceId' => null,
+            'identifierType' => null
+        ];
+
+        $key = new Key($this->projectId, $options);
+        $key->pathElement($kind, $identifier, $options['identifierType']);
+
+        return $key;
+    }
+
+    /**
+     * Create multiple keys with the same configuration.
+     *
+     * When inserting multiple entities, creating a set of keys at once can be
+     * useful. By defining the Key's kind and any ancestors up front, and
+     * allowing Cloud Datastore to allocate IDs, you can be sure that your
+     * entity settings are correct and that there will be no collisions during
+     * the insert operation.
+     *
+     * By default, IDs will be allocated to each key, requiring a service
+     * request. This can be disabled in the method options. Note that only a
+     * single service request will be executed, regardless of the number of keys
+     * generated, up to 500 keys (due to limits in Cloud Datastore).
+     *
+     * Example:
+     * ```
+     * $keys = $datastore->keys('Person', [
+     *     'number' => 10
+     * ]);
+     * ```
+     *
+     * ```
+     * // You can disable automatic allocation of IDs
+     * $keys = $datastore->keys('Person', [
+     *     'number' => 10,
+     *     'allocateIds' => false
+     * ]);
+     *
+     * echo $keys[0]->state(); // 'incomplete'
+     *
+     * $keys = $datastore->allocateIds($keys);
+     *
+     * echo $keys[0]->state(); // 'complete'
+     * ```
+     *
+     * ```
+     * // Ancestor paths can be specified
+     * $keys = $datastore->keys('Person', [
+     *     'ancestors' => [
+     *         ['kind' => 'Person', 'name' => 'Grandpa Joe'],
+     *         ['kind' => 'Person', 'name' => 'Dad Mike']
+     *     ],
+     *     'number' => 3
+     * ]);
+     * ```
+     *
+     * @see https://cloud.google.com/datastore/reference/rest/v1beta3/Key Key
+     * @see https://cloud.google.com/datastore/reference/rest/v1beta3/Key#PathElement PathElement
+     *
+     * @param string $kind The kind to use in the final path element
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type array[] $ancestors An array of
+     *           [PathElement](https://cloud.google.com/datastore/reference/rest/v1/Key#PathElement) arrays. Use to
+     *           create [ancestor paths](https://cloud.google.com/datastore/docs/concepts/entities#ancestor_paths).
+     *     @type bool $allocateIds Whether to automatically allocate IDs in the Datastore service. Set to `true` by
+     *           default. If set to `true`, a service request will be executed.
+     *     @type string $namespaceId Partitions data under a namespace. Useful for
+     *           [Multitenant Projects](https://cloud.google.com/datastore/docs/concepts/multitenancy)
+     *     @type int $number The number of keys to generate
+     *     @type string|int $id The ID for the last pathElement
```

Comment:
WDYT about using the same logic `key` above does where it attempts to auto detect id/name and provide another option to specify the identifier type in case someone would like to override?